Created IA for chef.io

mirage-nl opened the pull request.

moollaza commented:
@mirage-nl thanks for your contribution!

Can you please fill out the [pull request template](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/pull_request_template_spice.md) and edit your initial comment. Once completed, we'll be sure to review your PR and give you some feedback.

Thanks!

moollaza commented on lib/DDG/Spice/Chefio.pm:
```diff
@@ -0,0 +1,25 @@
+use DDG::Spice;
+
+primary_example_queries "cookbook underscore";
+description "Shows an NPM package";
```
Whoops, looks like this needs to be updated :)

moollaza commented on lib/DDG/Spice/Chefio.pm:
```diff
@@ -0,0 +1,25 @@
+code_url "https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/Npm.pm";
+topics "sysadmin", "programming";
+category "programming";
+attribution github  => ['https://github.com/remixz', 'zachbruggeman'],
```
Looks like the rest of the Metadata also needs to be updated. Would you care to provide some attribution details so we can give you credit?

moollaza commented on share/spice/chefio/chefio.js:
```diff
@@ -0,0 +1,36 @@
+        }
+
+        Spice.add({
+            id: "test",
```
Please update the `id` to match the name of your callback, i.e. `chefio`

moollaza commented on share/spice/chefio/chefio.js:
```diff
@@ -0,0 +1,36 @@
+    "use strict";
+    env.ddg_spice_chefio = function(api_result){
+        if (!api_result || api_result.error) {
+            return Spice.failed('test');
```
"test" here should also be `chefio`

moollaza commented on t/Chefio.t:
```diff
@@ -0,0 +1,26 @@
+
+ddg_spice_test(
+    [qw( DDG::Spice::Chefio)],
+    # At a minimum, be sure to include tests for all:
```
You can remove these comments, they're mostly for you to read and understand

moollaza commented on t/Chefio.t:
```diff
@@ -0,0 +1,26 @@
+    # At a minimum, be sure to include tests for all:
+    # - primary_example_queries
+    # - secondary_example_queries
+    'example query' => test_spice(
```
Please update the tests to make sure they're real queries your Instant Answer will handle. Also be sure to add some queries which should not trigger your Instant Answer:

e.g "cheap cookbook" should probably not trigger this, but currently it will.

moollaza commented on lib/DDG/Spice/Chefio.pm:
```diff
@@ -0,0 +1,25 @@
+spice wrap_jsonp_callback => 1;
+
+handle remainder => sub {
+	return lc $_ if $_;
```
You should probably look for and strip out the word "chef" in case someone search for "chef perl cookbook"

moollaza commented on lib/DDG/Spice/Chefio.pm:
```diff
@@ -0,0 +1,25 @@
+attribution github  => ['https://github.com/remixz', 'zachbruggeman'],
+            twitter => ['https://twitter.com/zachbruggeman', 'zachbruggeman'];
+
+triggers startend => 'cookbook';
```
I think we should add "chef cookbook" and "cookbook chef" to the triggers so "cpan chef cookbook" works and then `chef` would not be in the remainder.

mirage-nl commented:
Woops didn't check all the data I copied, I'll change it ASAP.

mirage-nl commented:
[x] Added metadata and attribution information
[x] Wrote test file and added to t/ directory
[x] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[x] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 10

    - Windows 7
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 8
        [] IE 9
        [] IE 10

    - Windows XP
        [] IE 7
        [] IE 8

    - Mac OSX
        [x] Google Chrome
        [x] Firefox
        [] Opera
        [x] Safari

    - iOS (iPhone)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - iOS (iPad)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - Android
        [] Firefox
        [] Native Browser
        [] Google Chrome
        [] Opera

zekiel commented:
Hey @mirage-nl  where did you find out about DuckDuckHack?

mirage-nl commented:
Hey @zekiel, I found it when I did some research on duckduckgo. I thought it was a good way to learn more about the community and the project it self.

moollaza commented on share/spice/chefio/chefio.js:
```diff
@@ -0,0 +1,36 @@
+                sourceUrl: "http://chef.io"
+            },
+            normalize: function(item) {
+                var a = {
```
Just FYI, you can actually return an annonymous object, rather than creating a named one and immediately returning one:

```js
normalize: function(item) {
    return {
        url: ...,
        title: ....,
        subtitle: ...
    }
},
...
```

mirage-nl replied:
Yeah true, should read my code better next time when using a template. Sorry about that!

zekiel commented:
thanks @mirage-nl  :)

moollaza commented:
Hey @mirage-nl I see you closed this -- how come? Are you not interested in finishing it? We'd love to see it go live!

mirage-nl commented:
I'm still interested but I'll make a better version, now I know the rules
and the layout and what information you guys want. So soon I'll reopen this
one or make a new pull request with the better version. Only question is
can I run the test with the duckpan server like Travis does so I can check
if that part works? Or how do I do this?

moollaza commented:
@mirage-nl you're welcome to reopen and continue to push updates to this PR. That's usually how it goes so we can easily follow along with the code from start to finish.

>Only question is can I run the test with the duckpan server like Travis does so I can check if that part works? Or how do I do this?

To run a test file in `t/`, `cd` into the root of the zeroclickinfo-spice directory and run `prove -Ilib t/`. Alternatively you can run `duckpan test`.

mirage-nl commented:
Okay then I'll reopen it soon. Thanks now I can test my test
On Feb 24, 2015 7:18 PM, "Zaahir Moolla" <notifications@github.com> wrote:

> @mirage-nl <https://github.com/mirage-nl> you're welcome to reopen and
> continue to push updates to this PR. That's usually how it goes so we can
> easily follow along with the code from start to finish.
>
> Only question is can I run the test with the duckpan server like Travis
> does so I can check if that part works? Or how do I do this?
>
> To run a test file in t/, cd into the root of the zeroclickinfo-spice
> directory and run prove -Ilib t/. Alternatively you can run duckpan test.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-spice/pull/1512#issuecomment-75813517>
> .
>

moollaza commented:
@mirage-nl just checking if you're still planning to open a new PR for this? Absolutely no rush, just curious :smiley:

mirage-nl commented:
Yeah I am but I'm waiting for a holiday because I'm kinda busy with my
internship.

On Fri, Mar 20, 2015, 5:43 AM Zaahir Moolla <notifications@github.com>
wrote:

> @mirage-nl <https://github.com/mirage-nl> just checking if you're still
> planning to open a new PR for this? Absolutely no rush, just curious [image:
> :smiley:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-spice/pull/1512#issuecomment-83903274>
> .
>

moollaza commented:
@mirage-nl Ahh alright, that's no problem! Marking this as `on hold` for now :)
